src/Articles.jsx

```diff
@@ -4,22 +4,86 @@ import { Link, useSearchParams } from "react-router-dom";
 function Articles() {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-  const [searchParams] = useSearchParams();
+  const [searchParams, setSearchParams] = useSearchParams();
   const topic = searchParams.get("topic");
+  const sortby = searchParams.get("sortby");
+  const orderby = searchParams.get("orderby");
+
+  function sortBy(query) {
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set("sortby", query);
+    setSearchParams(newParams);
+  }
+  function orderBy(order) {
+    const newOrderBy = new URLSearchParams(searchParams);
+    newOrderBy.set("orderby", order);
+    setSearchParams(newOrderBy);
+  }
+
   useEffect(() => {
     setIsLoading(true);
-    getArticles(topic).then((articleData) => {
+    getArticles(topic, sortby, orderby).then((articleData) => {
       setArticles(articleData);
       setIsLoading(false);
     });
-  }, [topic]);
+  }, [topic, sortby, orderby]);
 
   return isLoading ? (
     <p>Loading Articles...</p>
   ) : (
     <section>
+      orderby:
```

Comment:
This seems to be working nice but would say the UI is a tad ambiguous. Maybe one to do if styling later on but visually it isnt clear which sortby/orderby buttons have been clicked. Would a different html component be better suited than a button?